WordPress/wordpress-playground
# wp-now: Add `executeWPCli()` function to download and execute WP-CLI

## What?

- Add the `executeWPCli` function to download and execute `wp-cli. 
- In the future, we may include the command `wp-now wp`. Currently, we drop it out until we improve the pthreads execution. Currently a PR in progress: https://github.com/WordPress/wordpress-playground/pull/346
- Surface `emscriptenOptions` to catch print and print error for `wp-cli` execution.

## Why?



- See https://github.com/WordPress/playground-tools/issues/25

## How?



It downloads the wp-cli.phar file if the file doesn't exist, then uses `php.cli()` to execute it.
There are some limitations in the `wp-cli` features. Some of them may not work.

## Testing Instructions


-  Check out this branch.
- Copy your path to your theme or plugin
- After installing and building the project, run:
- Run the tests `npx nx test wp-now`
- Observe the tests pass.
